winrm/client.go

```diff
@@ -23,21 +29,38 @@ type Client struct {
 
 // NewClient will create a new remote client on url, connecting with user and password
 // This function doesn't connect (connection happens only when CreateShell is called)
-func NewClient(endpoint *Endpoint, user, password string) (client *Client, err error) {
+func NewClient(endpoint *Endpoint, user, password string, auth AuthType) (client *Client, err error) {
```

Comment:
If they are mutually exclusive, then if the user provides a certificate we should do CertAuth otherwise attempt basic HTTP Auth. I really think we can infer if we want to do CertAuth by just looking at the provided parameters, and error if something is wrong (ie no cert or key but also no user/pass, etc...).

What do you think?